Do not try to load recipes for a simulated environment

vinayada1 opened the pull request:
# Description

Currently, for a simulated env, if a recipe if not found in the env, the deployment fails. This is unnecessary for simulated environments as there are no output resources rendered i.e. the result with or without loading the recipe is the same. Therefore, skip the loading of recipes in the case of simulated environments

## Type of change

<!--

Please select **one** of the following options that describes your change and delete the others. Clearly identifying the type of change you are making will help us review your PR faster, and is used in authoring release notes.

If you are making a bug fix or functionality change to Radius and do not have an associated issue link please create one now. 

-->

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
<!--

Please update the following to link the associated issue. This is required for some kinds of changes (see above).

-->

Fixes: #7156

vinayada1 committed:
Do not try to load recipes for a simulated environment

Signed-off-by: vinayada1 <28875764+vinayada1@users.noreply.github.com>

kachawla commented on pkg/recipes/engine/engine.go:
```diff
@@ -119,14 +128,18 @@ func (e *engine) Delete(ctx context.Context, opts DeleteOptions) error {
 	}
 
-	configuration, err := e.options.ConfigurationLoader.LoadConfiguration(ctx, recipe)
+	if configuration.Simulated {
```
A log here will be helpful as well for debugging, similar to what you have above in execute.

kachawla commented on pkg/recipes/driver/bicep.go:
```diff
@@ -126,11 +126,6 @@ func (d *bicepDriver) Execute(ctx context.Context, opts ExecuteOptions) (*recipe
 		logger.Info("using Azure provider", "deploymentID", deploymentID, "scope", providerConfig.Az.Value.Scope)
 	}
 
-	if opts.Configuration.Simulated {
```
We should delete this unit test https://github.com/radius-project/radius/blob/27c1633311f5128e9ac44b143e63761f85f6000a/pkg/recipes/driver/bicep_test.go#L381? Doesn't look like it was doing what we expected it to, since it isn't failing now. :) Same for terraform driver.

vinayada1 replied:
These tests are actually failing which is good :) But makes sense to remove them now

kachawla commented on pkg/recipes/engine/engine.go:
```diff
@@ -71,14 +72,22 @@ func (e *engine) Execute(ctx context.Context, opts ExecuteOptions) (*recipes.Rec
 
-	configuration, err := e.options.ConfigurationLoader.LoadConfiguration(ctx, recipe)
+	// No need to try executing the recipe if it's a simulated environment.
+	if configuration.Simulated {
```
Let's add unit test for this..

kachawla reviewed (commented):
Looks great! Couple of quick comments.

Do we have any functional tests for simulated environments scenarios or plan to add any?

vinayada1 committed:
fix comments

Signed-off-by: vinayada1 <28875764+vinayada1@users.noreply.github.com>

kachawla reviewed (commented):
Thanks for addressing the feedback.

* Is there a blocker that prevented us from adding unit tests for: https://github.com/radius-project/radius/blob/ed54654b6b46c3684b9d6264b7ebc2c4f7a2c4dd/pkg/recipes/configloader/environment.go#L114
(Not directly related to this PR, but related to simulated env test coverage and the code this PR is consuming)

* Do we have any functional tests for simulated environments scenarios or plan to add any?

vinayada1 committed:
fix comments - add ut

Signed-off-by: vinayada1 <28875764+vinayada1@users.noreply.github.com>

vinayada1 commented:
> Thanks for addressing the feedback.
> 
> * Is there a blocker that prevented us from adding unit tests for: https://github.com/radius-project/radius/blob/ed54654b6b46c3684b9d6264b7ebc2c4f7a2c4dd/pkg/recipes/configloader/environment.go#L114
                   Thanks for pointing out. Added one now.
>   
>   (Not directly related to this PR, but related to simulated env test coverage and the code this PR is consuming)
> * Do we have any functional tests for simulated environments scenarios or plan to add any?
                   test/functional/datastoresrp/resources/simulated_environment_test.go has some tests. There are UTs too.
